jetstream/stream_config.go

```diff
@@ -26,154 +26,356 @@ import (
+		// the stream will retain.
+		MaxMsgsPerSubject int64 `json:"max_msgs_per_subject"`
+
+		// MaxMsgSize is the maximum size of any single message in the stream.
+		MaxMsgSize int32 `json:"max_msg_size,omitempty"`
+
+		// Storage specifies the type of storage backend used for the stream
+		// (file or memory).
+		Storage StorageType `json:"storage"`
+
+		// Replicas is the number of stream replicas in clustered JetStream.
+		// Defaults to 1, maximum is 5.
+		Replicas int `json:"num_replicas"`
+
+		// NoAck is a flag to disable acknowledging messages received by this
+		// stream.
+		NoAck bool `json:"no_ack,omitempty"`
+
+		// Duplicates is the window within which to track duplicate messages.
+		// Defaults to 0.
+		Duplicates time.Duration `json:"duplicate_window,omitempty"`
+
+		// Placement is used to declare where the stream should be placed via
+		// tags and/or an explicit cluster name.
+		Placement *Placement `json:"placement,omitempty"`
+
+		// Mirror defines the configuration for mirroring another stream.
+		Mirror *StreamSource `json:"mirror,omitempty"`
+
+		// Sources is a list of other streams this stream sources messages from.
+		Sources []*StreamSource `json:"sources,omitempty"`
+
+		// Sealed streams do not allow messages to be published or deleted via limits or API,
+		// sealed streams can not be unsealed via configuration update. Can only
+		// be set on already created streams via the Update API.
+		Sealed bool `json:"sealed,omitempty"`
+
+		// DenyDelete restricts the ability to delete messages from a stream via
+		// the API. Defaults to false.
+		DenyDelete bool `json:"deny_delete,omitempty"`
+
+		// DenyPurge restricts the ability to purge messages from a stream via
+		// the API. Defaults to false.
+		DenyPurge bool `json:"deny_purge,omitempty"`
+
+		// AllowRollup allows the use of the Nats-Rollup header to replace all
+		// contents of a stream, or subject in a stream, with a single new
+		// message.
+		AllowRollup bool `json:"allow_rollup_hdrs,omitempty"`
+
+		// Compression specifies the message storage compression algorithm.
+		// Defaults to NoCompression.
+		Compression StoreCompression `json:"compression"`
+
+		// FirstSeq is the initial sequence number of the first message in the
+		// stream.
+		FirstSeq uint64 `json:"first_seq,omitempty"`
+
+		// SubjectTransform allows applying a transformation to matching
+		// messages' subjects.
 		SubjectTransform *SubjectTransformConfig `json:"subject_transform,omitempty"`
 
-		// Allow republish of the message after being sequenced and stored.
+		// RePublish allows immediate republishing a message to the configured
+		// subject after it's stored.
 		RePublish *RePublish `json:"republish,omitempty"`
 
-		// Allow higher performance, direct access to get individual messages. E.g. KeyValue
+		// AllowDirect enables direct access to individual messages using direct
+		// get API. Defaults to false.
```

Comment:
we should consider flipping this default to true now

Reply:
I think we can do that in the next minor release.